src/Components/CreateImageWizard/validators.ts

```diff
@@ -57,6 +57,24 @@ export const isFileSystemConfigValid = (partitions: Partition[]) => {
   return duplicates.length === 0;
 };
 
+export const isUserNameValid = (userName: string) => {
+  if (userName === undefined) return false;
+  const isLengthValid = userName.length <= 32;
+  const isNotNumericOnly = !/^\d+$/.test(userName);
+  const isPatternValid = /^[a-zA-Z0-9][a-zA-Z0-9_.-]*[a-zA-Z0-9_$]$/.test(
+    userName
+  );
+
+  return isLengthValid && isNotNumericOnly && isPatternValid;
+};
+
+export const isSshKeyValid = (sshKey: string) => {
+  const isLengthValid = sshKey !== undefined && sshKey.length >= 2;
+  const isPatternValid =
+    /^(ssh-(rsa|dss|ed25519)|ecdsa-sha2-nistp(256|384|521)) \S+/.test(sshKey);
```

Comment:
Do you have any source for the regex? If so, could you please add it to a comment so we can reference it if needed?

I wonder if we should try do to a full match instead of a partial one, should be safer that way. We would need to add a few more things to allow for the label part of the key. So maybe something like:
```
^(ssh-(rsa|dss|ed25519)|ecdsa-sha2-nistp(256|384|521))\s+[A-Za-z0-9+\/=]+\s+\S+$
```

What do you think?

Reply:
I tried this  ```
 /^(ssh-(rsa|dss|ed25519)|ecdsa-sha2-nistp(256|384|521))\s+[A-Za-z0-9+=]+\s+\S+/.test(
      sshKey
    );
```
and the value 'ssh-rsa d' isnt valid
I took this pattern from 'edge' , they also check the ssh key validation

Reply:
Hmm, comment might be optional isn't it :thinking: this should work then
```
^(ssh-(rsa|dss|ed25519)|ecdsa-sha2-nistp(256|384|521))\s+[A-Za-z0-9+\/=]+(\s+\S+)?$
```

Reply:
do you think about this comment?
  // 1. Key types: ssh-rsa, ssh-dss, ssh-ed25519, or ecdsa-sha2-nistp(256|384|521).
  // 2. Base64-encoded key material.
  // 3. Optional comment at the end.